src/core/init/duration_updater.ts

```diff
@@ -42,25 +89,48 @@ export default function DurationUpdater(
 ) : Observable<never> {
   return fromEvent(manifest, "manifestUpdate").pipe(
     startWith(null),
-    tap(() => {
-      const maximumPosition = manifest.getMaximumPosition();
-
-      // Some targets poorly support setting a very high number for durations.
-      // Yet, in live contents, we would prefer setting a value as high as possible
-      // to still be able to seek anywhere we want to (even ahead of the Manifest if
-      // we want to). As such, we put it at a safe default value of 2^32 excepted
-      // when the maximum position is already relatively close to that value, where
-      // we authorize exceptionally going over it.
-      const newDuration = !manifest.isLive ? maximumPosition :
-                                             Math.max(Math.pow(2, 32),
-                                                      maximumPosition + YEAR_IN_SECONDS);
+    scan((acc) => {
+      return [acc[1], manifest.getMaximumPosition()] as [number, number];
+    }, [null, manifest.getMaximumPosition()] as [null | number, number]),
+    switchMap(([oldMaximumPosition, maximumPosition]) => {
+      return mediaSourceIsOpened$(mediaSource).pipe(
+        switchMap((isMediaSourceOpened) => {
+          if (!isMediaSourceOpened) {
+            return EMPTY;
+          }
+          return interval(100).pipe(
+            filter(() => checkForUpdatingSourceBuffer(mediaSource.sourceBuffers)),
+            timeout(1000)
```

Comment:
Why timeouting at all?

Reply:
I didn't want to keep the interval alive after 1 second

Reply:
But it is obsolete now

Reply:
It's not (I indicated that I think we should keep both the timeout and the update). 

Also I think only the OP should resolve the conversation